packages/block-editor/src/components/block-list/breadcrumb.js

```diff
@@ -21,12 +25,27 @@ import BlockTitle from '../block-title';
  *
  * @return {WPComponent} The component to be rendered.
  */
-function BlockBreadcrumb( { clientId, ...props } ) {
-	const ref = useRef();
+function BlockBreadcrumb( { clientId, rootClientId, moverDirection, ...props } ) {
+	function selector( select ) {
+		const {
+			__unstableGetBlockWithoutInnerBlocks,
+			getBlockIndex,
+		} = select( 'core/block-editor' );
+		const index = getBlockIndex( clientId, rootClientId );
+		const { name, attributes } = __unstableGetBlockWithoutInnerBlocks( clientId );
+		return { index, name, attributes };
+	}
+
+	const {
+		index,
+		name,
+		attributes,
+	} = useSelect( selector, [ clientId, rootClientId ] );
```

Comment:
I still don't like this pattern for useSelect personally, it makes it very hard to understand the relationship between the deps and the selector and whether we forgot a dep or not.

Reply:
What would you do instead rename the constants?

Reply:
This becomes a bit more succinct with the selector version of `getAccessibleBlockLabel` in #19664:

https://github.com/WordPress/gutenberg/pull/19664/files#diff-1f379cb4d47b0fe36bd77d3b6e1f4b60R26-R30